src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerAncestorFingerprintFacet.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.annotation.Nonnull;
+
+/**
+ * Facet representing the fact that some Docker image was derived from this one.
+ * This facet would be added to the fingerprint corresponding to the {@code FROM} instruction in a {@code Dockerfile}.
+ * {@link #getDescendantImageId} indicates the image that was built from that file.
+ * @see DockerFingerprints#addFromFacet
+ * @see DockerDescendantFingerprintFacet
+ */
+public class DockerAncestorFingerprintFacet extends DockerRunPtrFingerprintFacet {
```

Comment:
`DockerAncestorOfFingerprintFacet` does not read well. It sounds like something is an ancestor of the fingerprint!

I have reread this several times but still find the original names sounding right to me.